fingerprint2.js

```diff
@@ -21,1431 +21,1376 @@
+      obj.forEach(iterator)
+    } else if (obj.length === +obj.length) {
+      for (var i = 0, l = obj.length; i < l; i++) {
+        iterator(obj[i], i, obj)
+      }
+    } else {
+      for (var key in obj) {
+        if (obj.hasOwnProperty(key)) {
+          iterator(obj[key], key, obj)
         }
       }
-      this.userAgentKey(keys)
-      this.languageKey(keys)
-      this.colorDepthKey(keys)
-      this.deviceMemoryKey(keys)
-      this.pixelRatioKey(keys)
-      this.hardwareConcurrencyKey(keys)
-      this.screenResolutionKey(keys)
-      this.availableScreenResolutionKey(keys)
-      this.timezoneOffsetKey(keys)
-      this.timezoneKey(keys)
-      this.sessionStorageKey(keys)
-      this.localStorageKey(keys)
-      this.indexedDbKey(keys)
-      this.addBehaviorKey(keys)
-      this.openDatabaseKey(keys)
-      this.cpuClassKey(keys)
-      this.platformKey(keys)
-      this.doNotTrackKey(keys)
-      this.pluginsKey(keys)
-      this.canvasKey(keys)
-      this.webglKey(keys)
-      this.webglVendorAndRendererKey(keys)
-      this.adBlockKey(keys)
-      this.hasLiedLanguagesKey(keys)
-      this.hasLiedResolutionKey(keys)
-      this.hasLiedOsKey(keys)
-      this.hasLiedBrowserKey(keys)
-      this.touchSupportKey(keys)
-      this.customEntropyFunction(keys)
-      this.fontsKey(keys, function (keysWithFont) {
-        that.audioKey(keysWithFont, function (newKeys) {
-          that.enumerateDevicesKey(newKeys, function (keysWithDevices) {
-            var values = []
-            that.each(keysWithDevices.data, function (pair) {
-              var value = pair.value
-              if (value && typeof value.join === 'function') {
-                values.push(value.join(';'))
-              } else {
-                values.push(value)
-              }
-            })
-            var murmur = that.x64hash128(values.join('~~~'), 31)
-            return done(murmur, keysWithDevices.data)
-          })
-        })
-      })
-    },
-    // https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices/enumerateDevices
-    enumerateDevicesKey: function (keys, done) {
-      if (this.options.excludeEnumerateDevices || !this.isEnumerateDevicesSupported()) {
-        return done(keys)
+    }
+  }
+
+  var map = function (obj, iterator) {
+    var results = []
+    // Not using strict equality so that this acts as a
+    // shortcut to checking for `null` and `undefined`.
+    if (obj == null) {
+      return results
+    }
+    if (Array.prototype.map && obj.map === Array.prototype.map) { return obj.map(iterator) }
+    each(obj, function (value, index, list) {
+      results.push(iterator(value, index, list))
+    })
+    return results
+  }
+
+  var extendSoft = function (target, source) {
+    if (source == null) { return target }
+    var value
+    var key
+    for (key in source) {
+      value = source[key]
+      if (value != null && !target[key]) {
```

Comment:
target[key] should be checked again undefined or similar.
As currently 'extendSoft({detectScreenOrientation: false}, defaultOptions)' would keep 'detectScreenOrientation' set to true.

Reply:
good, Object.prototype.hasOwnProperty works for everything, it would not overwrite explicit undefined for example